lib/plugins/nwc/src/lib.rs

```diff
@@ -330,11 +433,55 @@ impl SdkNwcService {
         info!("Sending event: {event:?}");
         ctx.event_manager.notify(event).await;
     }
+
+    fn connections_have_changed(
+        active_connections: &mut HashMap<String, ActiveConnection>,
+    ) -> bool {
+        let now = utils::now();
+        let mut to_delete = vec![];
+        let mut updated = false;
+        for (name, ActiveConnection { connection, .. }) in active_connections.iter_mut() {
+            // If the connection has expired, mark it for deletion
+            if let Some(expiry) = connection.expiry_time_sec {
+                if now >= connection.created_at + expiry {
+                    to_delete.push(name.clone());
+                }
+            }
+            // If the connection's periodic budget has to be updated
+            if let Some(ref mut budget) = connection.periodic_budget {
+                if now >= budget.updated_at + budget.reset_time_sec {
+                    budget.used_budget_sat = 0;
+                    budget.updated_at = now;
+                    updated = true;
+                }
+            }
+        }
+        for name in &to_delete {
+            active_connections.remove(name);
+        }
+        !to_delete.is_empty() || updated
+    }
+
+    fn new_maybe_interval(ctx: &RuntimeContext) -> Option<Interval> {
+        ctx.persister
+            .get_min_interval()
+            .map(|interval| tokio::time::interval(Duration::from_secs(interval)))
+    }
+
+    async fn min_refresh_interval(maybe_interval: &mut Option<Interval>) -> Option<()> {
+        match maybe_interval {
+            Some(interval) => {
+                interval.tick().await;
+                Some(())
+            }
+            None => None,
+        }
+    }
```

Comment:
We should test this carefully on Wasm node.js. I don't recall the details but according to #880 it may cause panics.